x509/pkcs8.go

```diff
@@ -94,9 +102,32 @@ func MarshalPKCS8PrivateKey(key interface{}) ([]byte, error) {
 			return nil, errors.New("x509: failed to marshal EC private key while building PKCS#8: " + err.Error())
 		}
 
+	case ed25519.PrivateKey:
+		privKey.Algo = pkix.AlgorithmIdentifier{Algorithm: OIDPublicKeyEd25519}
+		privKey.PrivateKey = MarshalEd25519PrivateKey(k)
+
 	default:
 		return nil, fmt.Errorf("x509: unknown key type while marshalling PKCS#8: %T", key)
 	}
 
 	return asn1.Marshal(privKey)
 }
+
+// MarshalEd25519PrivateKey converts an Ed25519 private key to ASN.1 DER encoded form
+// (as an OCTET STRING holding the key seed).
+func MarshalEd25519PrivateKey(key ed25519.PrivateKey) []byte {
+	b, _ := asn1.Marshal(key.Seed())
+	return b
+}
+
+// ParseEd25519PrivateKey returns an Ed25519 private key from its ASN.1 DER encoded form.
+func ParseEd25519PrivateKey(der []byte) (ed25519.PrivateKey, error) {
+	var keySeed []byte
+	if _, err := asn1.Unmarshal(der, &keySeed); err != nil {
+		return nil, err
+	}
+	if len(keySeed) != ed25519.SeedSize {
+		return nil, fmt.Errorf("ed25519: bad seed length %d", len(keySeed))
```

Comment:
nit: Include the correct seed length in the error message
```suggestion
		return nil, fmt.Errorf("x509: ed25519 seed length should be %d bytes, got %d", ed25519.SeedSize, len(keySeed))
```

Reply:
Done.